### DISCUSSION TITLE
How come my dropdown inside the header works on large screens, but not for mobile responsive menu?

### DISCUSSION THREAD

**Discussion by AndresJaramillo95** [alpinejs/alpine#1347] [1. Help] [answered]
I'm trying to use AlpineJS to close and open dropdowns for my header NAV. The header is responsive for large and mobile screens. I'm able to get the header dropdown to work on large screens, but for some reason it does not work on mobile screens? I'm basically using the exact same syntax and tried many others. Still no luck. Maybe this is a bug? Could someone let me know if I'm in the wrong here?

[Here is the codesandbox](https://codesandbox.io/s/dreamy-rain-ljux4?file=/index.html:1390-1396) I'm using TailwindCSS and left everything in there for the most part to replicate my problem.

Please note my mobile menu button is hidden on large screens, its on line 42

The dropdown <div> that it's meant to open is on line 327

Any solution to this?

**Comment by KevinBatdorf** [comment, marked as answer]
Your button to open the mobile menu is outside the Alpine component (the `x-data`). You need to move it inside, or set up a second component on the button (add `x-data`) that fires an event to the window (`$dispatch('toggle-mobile-menu')`) that the mobile menu listens for (`@toggle-mobile-menu.window="mobileOpen = !mobileOpen"`)

```html
<button x-data @click="$dispatch('toggle-mobile-menu')"></button>
<!-- Elsewhere -->
<div x-data="{mobileOpen = false}" @toggle-mobile-menu.window="mobileOpen = !mobileOpen"></div>
```
